src/routes/controllers/channel.controller.js

```diff
@@ -0,0 +1,51 @@
+const mongoose = require("mongoose");
+const createError = require("http-errors");
+
+const Project = require("../../../models/Project");
+const Channel = require("../../../models/Channel");
+
+const { MESSAGE, LIMITED_CHANNEL_COUNT } = require("../../constants");
+
+const { LIMITED_CHANNEL, BAD_REQUEST } = MESSAGE;
+
+const createChannel = async (req, res, next) => {
+  const session = await mongoose.startSession();
+
+  try {
+    const { title, description } = req.body;
+    const { projectId } = req.params;
+    const channelId = mongoose.Types.ObjectId();
+
+    if (!mongoose.isValidObjectId(projectId)) {
+      return next(createError(400, BAD_REQUEST));
+    }
+
+    const selectedProject = await Project.findById(projectId)
+      .lean()
+      .populate("channels");
+
+    const channelCount = selectedProject.channels.length;
+    if (channelCount > LIMITED_CHANNEL_COUNT) {
+      return next(createError(423, LIMITED_CHANNEL));
+    }
+
+    await session.withTransaction(async () => {
+      const newChannel = { _id: channelId, title, description };
+
+      await Channel.create([newChannel], { session });
+      await Project.findByIdAndUpdate(
+        projectId,
+        { $push: { channels: channelId } },
+        { session },
+      );
+    });
+
+    return res.json({ id: channelId });
+  } catch (error) {
+    return next(error);
+  }
```

Comment:
```suggestion
  } finally {
    session.endSession();
  }
```

세션이 끝났다는걸 알려주지 않아도 저장이 잘 되나요...?
https://mongoosejs.com/docs/transactions.html
> 몽구스 transaction

Reply:
넵 테스트한 결과, 등록기능은 잘 작동합니다!
하지만 혹시 모를 상황을 대비해 공식문서에 따라 추가해놓도록 하겠습니다~